memstore/common/data_type.go

```diff
@@ -583,6 +583,9 @@ func ArrayValueFromArray(value []interface{}, dataType DataType) (interface{}, e
 // we can also support comma delimited string value as long as the item not contain comma
 // "11,12,13"
 func ArrayValueFromString(value string, dataType DataType) (interface{}, error) {
+	if len(value) == 0 || value == "null" {
```

Comment:
does null has to be all lower case?
maybe create a constant for null value string

Reply:
does null represent a null list or a list with one element which is null?

Reply:
for simplicity, will treat null list and empty list both as null for now

Reply:
null is used in json parser, and we have other place use it too. yeah, create a constant will be better.

Reply:
will add special null vector in another diff